File: iOS/TUILiveKit/Sources/Component/NetWorkInfo/State/NetWorkInfoState.swift
//
//  NetWorkInfo.swift
//  Pods
//
//  Created by ssc on 2025/5/9.
//
import RTCRoomEngine
#if canImport(TXLiteAVSDK_TRTC)
import TXLiteAVSDK_TRTC
#elseif canImport(TXLiteAVSDK_Professional)
import TXLiteAVSDK_Professional
#endif


struct NetWorkInfoState {
    var videoState: VideoState = .normal
    var videoResolution: Int = 720

    var audioState: AudioState = .normal
    var audioQuality: TUIAudioQuality = .default
    var volume: Int = 50

    var rtt: UInt32 = 0
    var upLoss: UInt32 = 0
    var downLoss: UInt32 = 0
    var uploadSpeed: UInt64 = 0
    var netWorkQuality: TUINetworkQuality = .excellent

    var deviceTemperature: Int = 0
    var showToast: Bool = false
}
enum AudioState {
    case mute
    case close
    case normal
    case exception
}

enum VideoState {
    case close
    case normal
    case exception
}

File: Flutter/livekit/lib/component/network_info/state/network_info_state.dart
import 'package:flutter/cupertino.dart';
import 'package:rtc_room_engine/rtc_room_engine.dart';

class NetWorkInfoState {
  ValueNotifier<VideoState> videoState = ValueNotifier(VideoState.normal);
  ValueNotifier<int> videoResolution = ValueNotifier(720);

  ValueNotifier<AudioState> audioState = ValueNotifier(AudioState.normal);
  ValueNotifier<TUIAudioQuality> audioQuality = ValueNotifier(TUIAudioQuality.audioProfileDefault);
  ValueNotifier<int> volume = ValueNotifier(50);

  ValueNotifier<int> rtt = ValueNotifier(0);
  ValueNotifier<int> upLoss = ValueNotifier(9);
  ValueNotifier<int> downLoss = ValueNotifier(0);
  ValueNotifier<int> uploadSpeed = ValueNotifier(0);
  ValueNotifier<TUINetworkQuality> networkQuality = ValueNotifier(TUINetworkQuality.qualityExcellent);

  ValueNotifier<int> deviceTemperature = ValueNotifier(0);
  ValueNotifier<bool> showToast = ValueNotifier(false);
}

enum AudioState { mute, close, normal, exception }

enum VideoState { close, normal, exception }
